geonode/base/api/permissions.py

```diff
@@ -129,6 +129,30 @@ def has_object_permission(self, request, view, obj):
             return False
 
 
+class IsOwnerOrAdmin(permissions.BasePermission):
```

Comment:
I see code duplication with `IsOwnerOrReadOnly`, can we please optimize a bit this by avoiding logic duplication?